JM/CF/Scripts/3_Game/CommunityFramework/Spline/CF_Spline.c

```diff
@@ -0,0 +1,264 @@
+class CF_Spline : Managed /*CF_ObservableCollection*/
+{
+	autoptr ScriptInvoker Update = new ScriptInvoker();
+	
+	private ref CF_SplinePoint m_Head;
+	private ref CF_SplinePoint m_Tail;
+
+	private CF_SplinePoint m_Current;
+	private float m_Time;
+	private float m_MaxTime;
+
+	private vector m_Position;
+	private vector m_Orientation;
+
+	private float m_Tension;
+	
+	void CF_Spline()
+	{
+		SetTension(0.0);
+	}
+
+	void Tick(float dt)
+	{
+		SetTime(m_Time);
+
+		if (!m_Current) return;
+
+		m_Time += dt * m_Current.GetSpeedCoef();
+
+		float delta = m_Time - m_Current.GetStart();
+		float length = m_Current.GetLength();
+		if (length <= 0.0) return;
+
+		m_Current.Get(delta / length, m_Position, m_Orientation);
+
+		if (m_Time > m_MaxTime) m_Time = m_MaxTime;
+	}
+
+	vector GetPosition()
+	{
+		return m_Position;
+	}
+
+	vector GetOrientation()
+	{
+		return m_Orientation;
+	}
+
+	CF_SplinePoint GetCurrent()
```

Comment:
*stupid math teacher joke*: `GetCurrent` what? Apples, bananas, black holes?

`GetCurrentPoint()` though I do not quite what point I am getting here? The interpolation result in the current time T or one of the anchor points that is currently closest? This needs a descriptive name, please.

Reply:
Can rename to `GetCurrentNode`

Reply:
No point should be accurate, since it is a point in space. As I suggested, rename it to `GetCurrentPoint()`.

Also, you did not respond to my question about what this actually is.